src/script.js

```diff
@@ -145,23 +145,18 @@ function fromASM (asm) {
   }))
 }
 
-function decompilePushOnly (script) {
-  var chunks = decompile(script)
+function toStack (chunks) {
+  chunks = decompile(chunks)
   typeforce(isPushOnly, chunks)
 
   return chunks.map(function (op) {
     if (Buffer.isBuffer(op)) return op
-    if (op === OPS.OP_0) return op
+    if (op === OPS.OP_0) return new Buffer(0)
 
-    return op
+    return scriptNumber.encode(op - OP_INT_BASE)
   })
 }
```

Comment:
It also leaves up for question when we decide to decode as script number compared to a `Buffer`...